### PR TITLE
fixes #3

### LINKED ISSUE #3: Create blog post list page (L)
- [x] create html structure
- [x] get back blog posts from database
- [x] manage pagination

### PR DESCRIPTION
**Pull request by fabianrupin195**
This PR add a the posts list with a pagination

### CODE REVIEW

File: .php_cs.cache [hunk 1]
```diff
@@ -0,0 +1 @@
...abase\/PaginatedQuery.php":2114693537,"src\/Framework\/Twig\/RouterTwigExtension.php":3462828505,"src\/Framework\/Twig\/PagerFantaExtension.php":3209515950,"src\/Framework\/ContentLengthListener.php":3384649855,"src\/Framework\/views\/demo.php":3275178994,"src\/Framework\/GoogleListener.php":234071326,"src\/Framework\/Renderer\/PHPRenderer.php":2811871165,"src\/Framework\/Renderer\/TwigRenderer.php":1169188033,"src\/Framework\/Renderer\/RendererInterface.php":2049203875,"src\/Framework\/ResponseEvent.php":1383892312,"src\/Framework\/Application.php":1965618289,"src\/Framework\/StringResponseListener.php":2045780411,"src\/routeCollection.php":3938991847,"src\/Frontend\/Blog\/Controller\/BlogController.php":634875057,"src\/Frontend\/Calendar\/views\/bye.php":2335599501,"src\/Frontend\/Calendar\/views\/hello.php":3971107103,"src\/Frontend\/Calendar\/Controller\/LeapYearController.php":984033998,"src\/Frontend\/Calendar\/Controller\/HelloController.php":4268294434,"src\/Frontend\/Calendar\/Controller\/ErrorController.php":3914457132,"src\/Frontend\/Calendar\/Model\/RenderTemplate.php":2951363868,"src\/Frontend\/Calendar\/Model\/LeapYear.php":2314203735,"src\/container.php":2553504606}}
```
Comment (aguidis) [inline .php_cs.cache]: add file `.php_cs.cache` in `.gitignore`

File: db/migrations/20180117094919_create_posts_table.php [hunk 2]
```diff
@@ -31,6 +31,7 @@ public function change()
     {
         $this->table('posts', ['collation' => 'utf8mb4_unicode_ci'])
             ->addColumn('title', MysqlAdapter::PHINX_TYPE_STRING)
+            ->addColumn('slug',MysqlAdapter::PHINX_TYPE_STRING)
```
Comment (aguidis) [inline db/migrations/20180117094919_create_posts_table.php]: add space after `,`

update `.php.cs`:

```
$config->getFinder()
    ->in([
        __DIR__.'/src',
       __DIR__.'/db',
    ])->exclude([
        'Tests'
    ]
);
```

so your migration files will be cleaned

File: db/seeds/PostSeeder.php [hunk 3]
```diff
@@ -20,17 +20,26 @@ public function run()
 
         for ($i = 0; $i < 100; $i++) {
             $date = $faker->unixTime('now');
+            $content = $faker->text(5000);
```
Comment (aguidis) [inline db/seeds/PostSeeder.php]: add a personal documentation in the method `run()`

File: src/Framework/ContentLengthListener.php [hunk 4]
```diff
@@ -18,6 +18,6 @@ public function onResponse(ResponseEvent $event)
 
     public static function getSubscribedEvents()
     {
-        return array('response' => array('onResponse',-255));
+        return ['response' => ['onResponse', -255]];
```
Comment (aguidis) [inline src/Framework/ContentLengthListener.php]: why did your set `-255` ?

File: src/Frontend/Blog/Controller/BlogController.php [hunk 5]
```diff
@@ -18,39 +19,50 @@ class BlogController
...
 
-    public function showPostsAction(Request $request)
+    public function showPostsAction(int $perPage, int $page)
     {
-        $posts = $this->pdo
-            ->query('SELECT * FROM posts ORDER BY created_at DESC LIMIT 10 ')
-            ->fetchAll();
+        $query = new PaginatedQuery(
+            $this->pdo,
+            'SELECT
```
Comment (aguidis) [inline src/Frontend/Blog/Controller/BlogController.php]: don't forget to use a repository later ;)

File: src/routeCollection.php [hunk 6]
```diff
@@ -5,22 +5,29 @@
...
 
-$routes->add('showPosts', new Route('/articles', array(
+$routes->add('showPosts', new Route('/articles/{page}', [
+    'perPage' => '5',
+    'page' => '1',
     '_controller' => 'Frontend\Blog\Controller\BlogController::showPostsAction',
-)));
+]));
 
+/*$routes->add('post', new Route('/article/{slug:[a-z\-0-9]+}-{id:[0-9]+}', array(
```
Comment (aguidis) [inline src/routeCollection.php]: if not used remove it

File: src/Framework/ContentLengthListener.php (hunk 4 above)
Comment (fabianrupin195) [reply src/Framework/ContentLengthListener.php]: To manage priorities of the listeners. I want this listener to be executed last.

**Pull request merged**